src/filter_tree/filter_tree.c

```diff
@@ -439,89 +434,109 @@ static inline void _FilterTree_In_Place_Set_Exp(FT_FilterNode *node, SIValue v)
 static bool _FilterTree_Compact_And(FT_FilterNode *node) {
 	// Try to compact left and right children.
 	bool is_lhs_const = FilterTree_Compact(node->cond.left);
-	bool ihs_rhs_const = FilterTree_Compact(node->cond.right);
+	bool is_rhs_const = FilterTree_Compact(node->cond.right);
 	// If both are not compactable, this node is not compactable.
-	if(!is_lhs_const && !ihs_rhs_const) return false;
-	// Compact to 'false' in case at least one child is 'false'
-	if(is_lhs_const) {
-		SIValue lhs_value = AR_EXP_Evaluate(node->cond.left->exp.exp, NULL);
+	if(!is_lhs_const && !is_rhs_const) return false;
+	// In every case from now, there will be a reduction, save the children in local placeholders for current node in-place modifications.
+	FT_FilterNode *lhs = node->cond.left;
+	FT_FilterNode *rhs = node->cond.right ;
+	// Both children are constants. This node can be set as constant expression.
+	if(is_lhs_const && is_rhs_const) {
+		// Both children are now contant expressions. We can evaluate and compact.
+		SIValue rhs_value = AR_EXP_Evaluate(rhs->exp.exp, NULL);
+		SIValue lhs_value = AR_EXP_Evaluate(lhs->exp.exp, NULL);
+		SIValue final_value = SI_BoolVal(SIValue_IsTrue(lhs_value) && SIValue_IsTrue(rhs_value));
```

Comment:
Add a comment: "Anding because we're reducing an AND node."

Reply:
done